service/test/BatchServerTest.cpp

```diff
@@ -74,20 +73,19 @@ class BatchServerTest : public ::testing::Test
         std::vector<geopm_request_s> m_signal_config;
         std::vector<geopm_request_s> m_control_config;
         std::shared_ptr<BatchServerImp> m_batch_server;
+        std::shared_ptr<BatchServerImp> m_batch_server_signals;
+        std::shared_ptr<BatchServerImp> m_batch_server_controls;
+        std::shared_ptr<BatchServerImp> m_batch_server_pid;
```

Comment:
What is the pro of having three different `BatchServerImp` objects, instead of having only one, whose definition changes based on the test fixture? It seems that only one of the three objects is used at any given time.

Reply:
This enables us to easily write more test fixtures that use any of these objects, and avoids issues around calling the destructor if we overwrite a the m_batch_server to set it to a new value.  It also allows the TearDown function to be responsible for more of the clean up after each test, thus reducing lines of code in each test fixture.